### PR TITLE
Add flag for dev and prod configuration

### LINKED ISSUE #9: Development server not working
The frontend currently attempts to connect to a web socket on the same port as the frontend is served on. If running the frontend development server, it attempts to connect to a web socket on this port (but the backend is on a different port). Some dev/prod flag should be added to address this.

### PR DESCRIPTION
**Pull request by isacsund**
- Add flag for getting the websocket url depending on the Node environment. 
- Rename PORT to SERVER_PORT for the server to avoid port conflicts with the client.
- Fixes #9

### CODE REVIEW

File: frontend/src/Constants.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+const prod = {
+  WS_URL: "ws://example.com"
+};
+
+const dev = {
+  WS_URL: "ws://localhost:" + (process.env.SERVER_PORT || 8080)
```
Comment (jhnj) [inline frontend/src/Constants.js:6]: "localhost" should be changed to `window.location.host`

File: frontend/src/Constants.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+const prod = {
+  WS_URL: "ws://example.com"
```
Comment (jhnj) [inline frontend/src/Constants.js:2]: In prod we want the websocket to connect to the same host and port that the site is served from, so: `"ws://" + window.location.host`

File: frontend/src/api.js [hunk 3]
```diff
@@ -8,8 +9,8 @@ export const useApi = (url) => {
   const [readyState, setReadyState] = useState("CONNECTING");
 
   useEffect(() => {
-    console.log(window.location.host);
-    socket.current = new WebSocket("ws://" + window.location.host);
+    console.log(config.WS_URL);
```
Comment (jhnj) [inline frontend/src/api.js:12]: I would add something like "WS_URL: " here or delete the log

File: frontend/src/Constants.js (hunk 1 above)
Comment (isacsund) [reply frontend/src/Constants.js:6]: Shouldn't this be `window.location.hostname`, since we just care about the hostname and not the port in this case?

Comment (isacsund) [reply frontend/src/Constants.js:6]: Another thing, `process.env.SERVER_PORT` is currently not recognized as CRA only defines environment variables starting with `REACT_APP_`.
Should we rename `SERVER_PORT` to `REACT_APP_SERVER_PORT` for both the backend and frontend? Is there a better solution to this?

**Pull request merged**